### ISSUE TITLE
wait=-1 should return 404

### ISSUE THREAD

**Issue by ghost** [zk1931/pulsefs#49]
wait=-1 should return 404 instead of 200.

```
curl "http://localhost:10000/hello?wait=-1" -v
> GET /hello?wait=-1 HTTP/1.1
> User-Agent: curl/7.32.0
> Host: localhost:10000
> Accept: */*
> 
< HTTP/1.1 200 OK
< Date: Mon, 08 Dec 2014 03:28:34 GMT
< Content-Length: 0
<
```

**Comment by EasonLiao** [comment]
If there's no node there we should return 404, how about the node gets deleted after the watch has been set? still return 404?

**Comment by ghost** [comment]
yup still 404

**Comment by EasonLiao** [comment]
How about delete without wait, just DELETE, I guess we can't return 404 since there would be no distinction between successful deletion and file not found?

**Comment by ghost** [comment]
A successful DELETE should return 200 (or maybe 204?). A DELETE on an non-existent file/directory is a 404.

http://stackoverflow.com/questions/2342579/http-status-code-for-update-and-delete

**Comment by EasonLiao** [comment]
not sure, then there's difference between normal deletion and deletion with
wait , if succeed normal deletion returns 200, wait deletion returns 404,
this is inconsistent

**Comment by ghost** [comment]
there is no delete with wait, only get with wait.

**Comment by EasonLiao** [comment]
Yeah you are right

**Comment by EasonLiao** [comment]
fixed in #50
